.github/workflows/codeql-analysis.yml

```diff
@@ -1,27 +1,61 @@
-name: CodeQL
+name: "CodeQL"
 
 on:
   push:
     branches:
       - main
 
   pull_request:
-    paths:
-      - .github/workflows/codeql-analysis.yml
     branches:
       - main
+    paths:
+      - .github/workflows/codeql-analysis.yaml
+      - "**/*.py"
+      - "**/*.js"
+      - "**/*.ts"
+      - requirements.txt
+      - requirements-dev.txt
+      - setup.cfg
+      - alembic.ini
```

Comment:
We did not do this before because it's super slow and blocks rapid development — do we need to run this on every pull request?

Reply:
Perhaps it's faster now?

Reply:
we don't _need_ to, but it's convenient because it'll show changes when people are making them, and flag new issues -- it's also not a required check so shouldn't block merges (people can merge without waiting for the analysis to finish, but if they happen to have a long-open PR then they'll benefit from the checks)

Reply:
👍 alrighty

Reply:
also if it becomes a problem again we can always adjust this; no biggie!